### DISCUSSION TITLE
[Question] Is it possible to change the mesh of the object after every environment step?

### DISCUSSION THREAD

**Discussion by chandramohanjagadeesan** [isaac-sim/IsaacLab#1075] [Q&A]
I am working on a continual Reinforcement learning project where I want more objects appearing or disappearing in the environment. Is there a way to make it possible in Isaaclab? Ideally I would like to change the mesh of the object from one time step to another. Thanks in advance for your reply

**Comment by garylvov** [comment]
I don't believe this is directly supported, but you can certainly emulate this behavior.

I'd say that the closest thing to what you want is: https://github.com/isaac-sim/IsaacLab/issues/186

Some hacky workarounds you could try:
Spawn all of the objects of interest for each environment, and then "teleport" the objects you don't care about as far from the robot (say, 10m), and the objects you care about close to the robot.

**Reply by chandramohanjagadeesan** [reply]
Thanks for the reply. I am working on a next best view planning task using a robot. the agent does not know what object is present so I made a dome on top of the object where the object should not enter. So if the robot hits the dome there should be a negative reward. When the camera observes the dome and the object it will reassess the places where it cannot reach (ie the collision mesh would change). So I need the collision mesh of the object to be changed every time step. Is there a way to do this?. i have opened another issue #1062 which is for changing the object where your idea would work...
